pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1681,15 +2362,19 @@ func (cfg *configuration) Validate() error {
 	return nil
 }
 
-// gormToGRPCStatus takes an error, and converts it to a GRPC error.
-// If the error is a gorm error type with a known mapping to a GRPC
-// status, that code will be set, otherwise the code will be set to
-// Unknown.
+// gormToGRPCStatus takes an error, and converts it to a GRPC error.  If the
+// error is already a gRPC status , it will be returned unmodified. Otherwise
+// if the error is a gorm error type with a known mapping to a GRPC status,
+// that code will be set, otherwise the code will be set to Unknown.
 func gormToGRPCStatus(err error) error {
```

Comment:
nit: `errorToGRPCStatus` is probably a better name for the described functionality.

Reply:
In the interest of getting this change in, and keeping changes scoped, i'd rather leave any rename out. Tackle this in a follow-up?

Reply:
👍